nodes/mtnode.py

```diff
@@ -71,17 +74,41 @@ def __init__(self):
 
         self.frame_local = get_param('~frame_local', 'ENU')
 
-        self.diag_msg = DiagnosticArray()
+
         self.stest_stat = DiagnosticStatus(name='mtnode: Self Test', level=1,
                                            message='No status information')
         self.xkf_stat = DiagnosticStatus(name='mtnode: XKF Valid', level=1,
                                          message='No status information')
         self.gps_stat = DiagnosticStatus(name='mtnode: GPS Fix', level=1,
                                          message='No status information')
-        self.diag_msg.status = [self.stest_stat, self.xkf_stat, self.gps_stat]
+
+	self.device_info = {
+		'product_code': self.mt.GetProductCode(),
+                'device_id': self.mt.GetDeviceID(),
+                'firmware_rev': self.mt.GetFirmwareRev()
+        }
+
+        output_config = self.mt.GetConfiguration()
+        for (k,v) in output_config.iteritems():
+            self.device_info[k] = v
+
+        self.updater = diagnostic_updater.Updater()
+        self.updater.setHardwareID(str(self.mt.GetDeviceID()))
```

Comment:
From http://www.ros.org/reps/rep-0107.html

> If applicable this identifies the specific hardware running. This is for things like serial numbers for devices so that a piece of hardware can be tracked between robots if it is moved between robots or moved within a robot.